### PR TITLE
[linting_tool] Add lint rules list

### PR DESCRIPTION
**Pull request by abd99**
@domesticmouse, should I add a test for API calls using Mockito?

### CODE REVIEW

File: experimental/linting_tool/analysis_options.yaml [hunk 1]
```diff
@@ -3,7 +3,6 @@ include: package:flutter_lints/flutter.yaml
 analyzer:
   strong-mode:
     implicit-casts: false
-    implicit-dynamic: false
```
Comment (domesticmouse) [inline experimental/linting_tool/analysis_options.yaml:6]: I'm curious why this one is being deleted

File: experimental/linting_tool/lib/pages/home_page.dart [hunk 2]
```diff
@@ -3,13 +3,74 @@
...
 
-class HomePage extends StatelessWidget {
+class HomePage extends StatefulWidget {
   const HomePage({Key? key}) : super(key: key);
 
+  @override
+  _HomePageState createState() => _HomePageState();
+}
+
+class _HomePageState extends State<HomePage> {
```
Comment (domesticmouse) [inline experimental/linting_tool/lib/pages/home_page.dart:18]: I'm curious what state `_HomePageState` is holding on to. I don't see any fields here, nor a `setState`. What am I missing?

File: experimental/linting_tool/lib/pages/home_page.dart [hunk 3]
```diff
@@ -3,13 +3,74 @@
...
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance!.addPostFrameCallback((timeStamp) {
+      var rulesStore = Provider.of<RuleStore>(context, listen: false);
+      if (rulesStore.rules.isEmpty && !rulesStore.isLoading) {
+        rulesStore.fetchRules();
```
Comment (domesticmouse) [inline experimental/linting_tool/lib/pages/home_page.dart:25]: I'm curious why we have this `rulesStore.fetchRules()` load logic here specifically. It feels like something we should farm off to an isolate so we don't lag the main thread if CPU is a concern.

**Review by domesticmouse — APPROVED** [review]

**Comment by abd99** [comment]
PTAL, @domesticmouse. I have made the changes as per our discussion.

**Review by domesticmouse — APPROVED** [review]

**Comment by domesticmouse** [comment]
Hey @abd99, feel free to poke me on chat if I don't review your PRs in a timely manner =)

**Pull request merged**